Sources/Kitura/Router.swift

```diff
@@ -203,30 +203,61 @@ public class Router {
     internal func render(template: String, context: [String: Any],
                          options: RenderingOptions = NullRenderingOptions()) throws -> String {
         let (optionalFileExtension, resourceWithExtension) = calculateExtension(template: template)
+        
+        let templateEngine = try validateTemplate(template: template, optionalExtension: optionalFileExtension)
+        let absoluteFilePath = buildAbsoluteFilePath(with: resourceWithExtension)
+        
+        return try templateEngine.render(filePath: absoluteFilePath, context: context, options: options,
+                                         templateName: resourceWithExtension)
+    }
+    
+    /// Render a template using an Encodable type. 
+    /// - Parameter template: The path to the template file to be rendered.
+    /// - Parameter value: A value which conforms to Encodable to be used by the
+    ///                     template engine while rendering the template.
+    /// - Parameter forKey: A value used to match the Encodable value to the correct variable in a template file.
+    ///                                 The `forKey` value should match the desired variable in the template file.
+    /// - Parameter options: rendering options, specific per template engine
+    ///
+    /// - Returns: The content generated by rendering the template.
+    /// - Throws: Any error thrown by the Templating  Engine when it fails to
+    ///          render the template.
+    internal func render<T: Encodable>(template: String, with value: T, forKey key: String?,
+                         options: RenderingOptions = NullRenderingOptions()) throws -> String {
+        let (optionalFileExtension, resourceWithExtension) = calculateExtension(template: template)
+        
+        let templateEngine = try validateTemplate(template: template, optionalExtension: optionalFileExtension)
+        let absoluteFilePath = buildAbsoluteFilePath(with: resourceWithExtension)
+        
+        return try templateEngine.render(filePath: absoluteFilePath, with: value, forKey: key, options: options, templateName: resourceWithExtension)
+    }
+    
+    private func validateTemplate(template: String, optionalExtension: String?) throws -> TemplateEngine {
```

Comment:
As this returns a TemplateEngine a better name might be something like `getTemplateEngineForTemplate()`?